docs/v1/P4Runtime-Spec.mdk

```diff
@@ -2320,6 +2325,116 @@ following semantics.
 - `DELETE`: Delete the group entry and deallocate the `group_id`. The group
   should not be referenced in the table action of any table entry.
 
+### One Shot Action Selector Programming { #sec-oneshot }
+
+P4Runtime supports syntactic sugar to program a table, which is
+implemented with an action selector, in one shot. This means that a
+table entry, an action profile group, and a set of action profile
+members can programmed with a single update message. Using one shots
+has the advanatage that the controller does not need to keep track of
+group ids and member ids.
+
+One shots are programmed by choosing the `ActionProfileActions`
+message as the `TableAction`. The `ActionProfileActions` message
+consists of a set of `ActionProfileAction` messages, which in turn
+have the following fields:
+
+* `action` is one of the actions specified by the table that is being
+  programmed.
+
+* `weight` specifying the probability of the action's selection at
+  runtime.
+
+* `watch` is the controller defined 32-bit port number that the
+  action's liveness depends on. At runtime, the action must be
+  excluded from selection if the this watch port is down.
+
+Semantically, one shots are equivalent to programming the table entry,
+group, and members individually. The necessary group id and member ids
+are bound to unused ids.
+
+For example, consider again the action selector defined in the
+beginning of this section. This table could be programmed with the
```

Comment:
Maybe add a reference to the code block instead to be clearer (see http://madoko.org/reference.html#sec-ids-labels)